### ISSUE TITLE
Implement SEIR model class

### ISSUE THREAD

**Issue by DavAug** [SABS-R3-Epidemiology/seirmo#6]
Implement SEIR model class specified in [here](https://benlambertdotcom.files.wordpress.com/2020/10/week_project.docx).

- [x] Define the SEIR model class in the `models.py` called `SEIRModel` that inherits from #5.
- [x] Implement the simulate method for the SEIR model with inputs and outputs according to the base class doc string. Hint: You can choose any numerical integration package that you like.
- [x] Write a doc string which explains and defines the differential equations, see [here](https://github.com/pints-team/pints/blob/master/pints/toy/_fitzhugh_nagumo_model.py) for the raw doc string, and [here](https://pints.readthedocs.io/en/stable/toy/fitzhugh_nagumo_model.html) for the sphinx illustration.

**Comment by siting-miao** [comment]
Hey @DavAug , could you please help me have a look at my initial build up of the SEIRModel class? Which is pushed under issue-01 branch (sorry I forgot to switch). I'll add doc strings later, but I want to see whether what I have written makes sense.

**Comment by DavAug** [comment]
> Hey @DavAug , could you please help me have a look at my initial build up of the SEIRModel class? Which is pushed under issue-01 branch (sorry I forgot to switch). I'll add doc strings later, but I want to see whether what I have written makes sense.

Yes @siting-miao, no problem.

- [ ] Could you please define the SEIRModel class within the models.py? The idea is that the models.py contains all models of the repository.
The implementation looks really good but we could improve a little bit in terms of readability:
- [ ] Could we please factor out the `f` function to be a private method of the class, and also give it a slightly more descriptive name maybe `right_hand_side` or `rhs`. Then the method name would be, e.g. `_right_hand_side` (the `_` makes it private).
- [ ] The readability would improve a lot if you got the entries of `y` and `c` and gave them names that coincide with your docstring formula. Taking the names from your comment this would be
```
s, e, i, r = y
beta, kappa, gamma = c
```
- [ ] It would then make sense to break the `dydt` list for each dimension (again just for readability)
```
dydt = [
    -beta * s * i,
    beta * s * i - kappa * e,
    kappa * e - gamma * i,
    gamma * i
```
- [ ] You don't really need to assign `times` to `time_span`, but you could just keep using `time`.

But this looks really good! Just cosmetics.

Regarding your question about the times and the tuple expected from the SimulationController: that is on purpose and we can just chat about it tomorrow.

**Comment by siting-miao** [comment]
@DavAug Thank you so much for your detailed comments! All of them are really helpful. I have pushed all the changes. There seems to be one error in line 69 in the use of solve_ivp when I call the f function, and I am not sure how could I fix it. Could you please help me have a look when you have time?

**Comment by DavAug** [comment]
Another example for equations in doc strings 
https://github.com/pints-team/pints/blob/master/pints/_log_pdfs.py
https://pints.readthedocs.io/en/latest/log_pdfs.html#pints.PooledLogPDF

**Comment by siting-miao** [comment]
@DavAug I have written something for the docstring. I'm not sure how I could see the actual outcome of this docstring?

**Comment by DavAug** [comment]
We will have to setup Sphinx before we can do that, which is what Hui Jia is working on isn’t she? So I’ll just review the PR and if the docstring turns out to be weird, we fix it later in another issue.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
